.github/workflows/deploy.yml

```diff
@@ -0,0 +1,35 @@
+name: Deploy
+on:
+    push:
+        branches:
+            - cargo_generate_guide
+
+jobs:
+    deploy:
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v2
+              with:
+                  fetch-depth: 0
+            - name: Install mdbook
+              run: |
+                  mkdir mdbook
+                  curl -sSL https://github.com/rust-lang/mdBook/releases/download/v0.4.14/mdbook-v0.4.14-x86_64-unknown-linux-gnu.tar.gz | tar -xz --directory=./mdbook
```

Comment:
only for curiosity: why not depending on the binary via `cargo install mdbook`? 
At least that way it would always pull in the latest version of it.

Reply:
also maybe that simplifies a bit of custom code: [peaceiris/actions-mdbook](https://github.com/peaceiris/actions-mdbook)

Reply:
Switched to this action setup.